Dockerfile

```diff
@@ -1,78 +1,27 @@
-#**********************************************************************
-# Builder
-# Create a go runtime suitable for building and testing kfctl
-ARG GOLANG_VERSION=1.13.7
-FROM golang:$GOLANG_VERSION as builder
-
-ARG BRANCH=master
-ARG REPO=https://github.com/kubeflow/kubeflow
-
-RUN apt-get update
-RUN apt-get install -y git unzip jq vim
-
-# junit report is used to conver go test output to junit for reporting
-RUN go get -u github.com/jstemmer/go-junit-report
-
-# We need gcloud to get gke credentials.
-RUN if [ "$(uname -m)" = "x86_64" ]; then \
-        cd /tmp && \
-        wget -nv https://dl.google.com/dl/cloudsdk/release/install_google_cloud_sdk.bash && \
-        chmod +x install_google_cloud_sdk.bash && \
-        ./install_google_cloud_sdk.bash --disable-prompts --install-dir=/opt/; \
-    fi
-
-ENV PATH /go/bin:/usr/local/go/bin:/opt/google-cloud-sdk/bin:${PATH}
-
-# use go modules
-ENV GO111MODULE=on
-ENV GOPATH=/go
-# Workaround for https://github.com/kubernetes/gengo/issues/146
-ENV GOROOT=/usr/local/go
-
-# Create kfctl folder
-RUN mkdir -p ${GOPATH}/src/github.com/kubeflow/kfctl
-WORKDIR ${GOPATH}/src/github.com/kubeflow
-RUN mkdir kubeflow
-RUN echo REPO=${REPO} branch=${BRANCH}
-RUN git clone ${REPO} --depth=1 --branch ${BRANCH} --single-branch kubeflow
-WORKDIR ${GOPATH}/src/github.com/kubeflow/kfctl
-
-# Download dependencies first to optimize Docker caching.
-COPY go.mod .
-COPY go.sum .
+# Build the manager binary
+FROM golang:1.18 as builder
```

Comment:
Can we use one of the one of the ubi8/golang tool images?
I'm seeing a bunch at

```
❯ skopeo list-tags docker://registry.access.redhat.com/ubi8/go-toolset | grep 1.18
        "1.18",
        "1.18.4",
        "1.14.12-17.1618436992-source",
        "1.18.4-8.1669838000-source",
        "1.14.12-17.1618436992",
        "1.18.4-8-source",
        "1.18.4-8.1669838000",
        "1.18.4-8",
```

Reply:
Could you actually use registry.redhat.io/ubi8/go-toolset instead?

Reply:
Update the Dockerfile with `registry.redhat.io/ubi8/go-toolset` image with Go version as a docker `ARG`